### PR TITLE
Add coalaBear

### PR DESCRIPTION
**Pull request by Asnelchristian**
Closes https://github.com/coala-analyzer/coala-bears/issues/600

### CODE REVIEW

File: bears/general/coalaBear.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Spelling'}
+
+    def run(self, filename, file):
+        """
+        Check for the correct spelling of ``coala`` in the file.
+        """
+        for line_number, line in enumerate(file):
+            if line.find("Coala") != -1 or line.find('COALA') != -1:
```
Comment (sils1297) [inline bears/general/coalaBear.py]: hm, could we rather do a regex match and match an upper case c and then oala case insensitive?

CC @Makman2

File: bears/general/coalaBear.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+                yield Result.from_values(
+                    origin=self,
+                    message=("The spelling of ``coala`` is wrong in line {}."
+                             .format(line_number)),
+                    severity=RESULT_SEVERITY.NORMAL,
+                    file=filename)
```
Comment (sils1297) [inline bears/general/coalaBear.py]: can we have a diff autocorrecting the c to a lower case one?

File: tests/general/coalaBearTest.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+bad_file_1 = """
+Coala"""
+
+bad_file_2 = """
+COALA"""
+
+coalaBearTest = verify_local_bear(
+    coalaBear,
+    valid_files=(good_file,),
+    invalid_files=(bad_file_1, bad_file_2))
```
Comment (sils1297) [inline tests/general/coalaBearTest.py]: this is super awesome :)

File: bears/general/coalaBear.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import re
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.results.Diff import Diff
```
Comment (mr-karan) [inline bears/general/coalaBear.py]: not really sure whether this is a worthy change, but according to alphabetical imports, this line should go up.

File: bears/general/coalaBear.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Spelling'}
+
+    def run(self, filename, file):
+        """
+        Check for the correct spelling of ``coala`` in the file.
+        """
+        corrected = []
+        for line in file:
+            wrong_spelling = r'C([o,O][a,A][l, L][a,A])'
```
Comment (sils1297) [inline bears/general/coalaBear.py]: can't you simply do `[oO][aA]` instead of the comma thing?

File: bears/general/coalaBear.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            corrected += [re.sub(wrong_spelling,
+                                 lambda match: 'c' + match.group(1),
+                                 line)]
+        diffs = Diff.from_string_arrays(file, corrected).split_diff()
+        for diff in diffs:
+            yield Result(self,
+                         "``coala`` always starts with lowercase ``c``.",
```
Comment (sils1297) [inline bears/general/coalaBear.py]: The message definitely *must* be: `coala is always written with a lower case c` because that's what we always write :)

**Comment by arafsheikh** [comment]
Isn't `cOALA `wrong too?

File: bears/general/coalaBear.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            corrected += [re.sub(wrong_spelling,
+                                 lambda match: 'c' + match.group(1),
+                                 line)]
+        diffs = Diff.from_string_arrays(file, corrected).split_diff()
+        for diff in diffs:
+            yield Result(self,
+                         "``coala`` is always written with a lower case ``c``",
+                         affected_code=(diff.range(filename),),
+                         diffs={filename: diff},
+                         severity=RESULT_SEVERITY.NORMAL)
```
Comment (arafsheikh) [inline bears/general/coalaBear.py]: According to https://github.com/coala-analyzer/coala-bears/issues/600 the severity should be `MAJOR`

**Comment by Asnelchristian** [comment]
nope it is not

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/628/commits/be28e915b2ee6849f987483994fb00151bf9ec8d

**Comment by sils1297** [comment]
can you rebase, reack, merge and make an asciinema?

**Comment by Asnelchristian** [comment]
okay :)


2016-07-25 23:07 GMT+01:00 Lasse Schuirmann <notifications@github.com>:

**Comment by sils1297** [comment]
https://github.com/coala-analyzer/coala-bears/pull/628/commits/b58bc8cc524d83765d06989e5dbd131f813157f0 s ready

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/628/commits/b58bc8cc524d83765d06989e5dbd131f813157f0

**Comment by Asnelchristian** [comment]
@rultor merge

**Pull request merged**